pkg/ingester/metrics.go

```diff
@@ -187,8 +195,20 @@ func newIngesterMetrics(r prometheus.Registerer, createMetricsConflictingWithTSD
 			Name: "cortex_ingester_inflight_push_requests",
 			Help: "Current number of inflight push requests in ingester.",
 		}, func() float64 {
-			if inflightRequests != nil {
-				return float64(inflightRequests.Load())
+			if inflightPushRequests != nil {
+				return float64(inflightPushRequests.Load())
+			}
+			return 0
+		}),
+
+		inflightQueryRequests: promauto.With(r).NewGaugeFunc(prometheus.GaugeOpts{
+			Name: "cortex_ingester_max_inflight_query_requests",
+			Help: "Max number of inflight query requests in ingester.",
+		}, func() float64 {
+			if maxInflightQueryRequests != nil {
+				r := maxInflightQueryRequests.Load()
+				maxInflightQueryRequests.Store(0)
```

Comment:
This is something I am not sure about. Do we want to reset this? Imagining `/metrics` API gets called twice within 1s, I expect both return similar results rather than 1 API I got a regular max inflight requests and another response I got 0.

For users who run HA prometheus, I think this is a problem.

Reply:
ok.. good call.. lemme change this.